src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import api from "./api";
+
+const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setLoading(true);
+    api.get(`/articles/${article_id}`)
```

Comment:
I'd consider refactoring this so you have all your API calls in a separate file, ie- api.js. Have a look at these notes for creating a base URL:

https://notes.northcoders.com/courses/js-front-end/react-router